File: Task12/Plant.m
//
//  Plant.m
//  Task12
//
//  Created by Sergey Plotnikov on 08.07.16.
//  Copyright © 2016 Sergey Plotnikov. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "Plant.h"


@interface Plant()

@property(nonatomic) NSString* name;
@property(nonatomic) NSInteger calories;

@end


@implementation Plant

@synthesize name = _name;
@synthesize calories = _calories;

- (instancetype)initWithName:(NSString *)aName
{
    self = [super init];
    
    if (self) {
        _name = aName;
        _calories = kDefaultPlantCalories;
    }
    
    return self;
}

- (NSString *)description
{
    return [NSString stringWithFormat:@"%@(%ld)", self.name, self.calories];
}

@end

File: Task1_2/Plant.swift
//
//  Plant.swift
//  Task1_2
//
//  Created by Admin on 22/09/16.
//  Copyright © 2016 Admin. All rights reserved.
//

import Foundation

class Plant: Creature, Nameable {

    private (set) var name: String

    override var description: String {
        get {
            return "Plant " + name
        }
    }
    
    init(name: String) {
        self.name = name
        super.init(calories: Creature.defaultPlantCalories)
    }

}
